src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -20,20 +20,21 @@ public partial class Ping
         private const int MaxIpHeaderLengthInBytes = 60;
         private static bool SendIpHeader => OperatingSystem.IsMacOS();
         private static bool NeedsConnect => OperatingSystem.IsLinux();
+        private static bool IsOSXLike = OperatingSystem.IsMacOS() || OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsWatchOS();
```

Comment:
This will not work on watchOS (lack of Sockets support). Instead, we should mark the whole assembly to throw PNSE on watchOS